nautobot/ipam/querysets.py

```diff
@@ -247,39 +247,62 @@ def net_contains_or_equals(self, prefix):
             broadcast__gte=last_ip,
         )
 
-    def get(self, *args, prefix=None, **kwargs):
+    def get(self, *args, **kwargs):
         """
         Provide a convenience for `.get(prefix=<prefix>)`
+
+        Only needed here (normally `get()` just redirects to `filter()` under the hood) because we specifically want
+        in the `get()` case (only) to *not* automatically convert `get(prefix="10.1.1.1/8")` to
+        `get(network="10.0.0.0", prefix_length=8)', i.e. discard the host bits.
+
+        TODO: why *shouldn't* we have similar enforcement on `filter()` and `exclude()`?
```

Comment:
It looks like this TODO was addressed in the code in the places `merge_dicts_without_collision` is being called? Is that accurate?

Reply:
I'm referring to the check at lines 265-266 in this function:

```python
            if str(_prefix) != str(prefix):
                raise self.model.DoesNotExist()
```

There's no equivalent check in the `filter()` method, so I didn't add it in the `exclude()` implementation either.